src/Components/Aspire.Keycloak/AspireKeycloakExtensions.cs

```diff
@@ -0,0 +1,103 @@
+
+        builder.AddJwtBearer(authenticationScheme);
+
+        builder.Services.AddHttpClient(KeycloakBackchannel);
+
+        builder.Services
+               .AddOptions<JwtBearerOptions>(authenticationScheme)
+               .Configure<IConfiguration, IHttpClientFactory, IHostEnvironment>((options, configuration, httpClientFactory, hostEnvironment) =>
+               {
+                   options.Backchannel = httpClientFactory.CreateClient(KeycloakBackchannel);
+                   options.Authority = GetAuthorityUri(serviceName, realm);
+
+                   configureJwtBearerOptions?.Invoke(options);
+               });
+    }
+
+    /// <summary>
+    /// Adds Keycloak OpenID Connect authentication to the application.
+    /// </summary>
+    /// <param name="builder">The <see cref="AuthenticationBuilder" /> to add services to.</param>
+    /// <param name="serviceName">The name of the service used to resolve the Keycloak server URL.</param>
+    /// <param name="realm">The realm of the Keycloak server to connect to.</param>
+    /// <param name="openIdConnectScheme">The OpenID Connect authentication scheme name. Default is "OpenIdConnect".</param>
+    /// <param name="cookieScheme">The cookie authentication scheme name. Default is "Cookie".</param>
+    /// <param name="configureOpenIdConnectOptions">An optional action to configure the <see cref="OpenIdConnectOptions"/>.</param>
+    /// <remarks>
+    /// The <paramref name="serviceName"/> is used to resolve the Keycloak server URL and is combined with the realm to form the authority URL.
+    /// For example, if <paramref name="serviceName"/> is "keycloak" and <paramref name="realm"/> is "myrealm", the authority URL will be "https+http://keycloak/realms/myrealm".
+    /// </remarks>
+    public static void AddKeycloakOpenIdConnect(
+        this AuthenticationBuilder builder,
+        string serviceName,
+        string realm,
+        string openIdConnectScheme = OpenIdConnectDefaults.AuthenticationScheme,
+        string cookieScheme = CookieAuthenticationDefaults.AuthenticationScheme,
+        Action<OpenIdConnectOptions>? configureOpenIdConnectOptions = null)
+    {
+        ArgumentNullException.ThrowIfNull(builder);
+
+        builder.AddCookie(cookieScheme)
+               .AddOpenIdConnect(openIdConnectScheme, options => { });
+
+        builder.Services.AddHttpClient(KeycloakBackchannel);
+
+        builder.Services
+               .AddOptions<OpenIdConnectOptions>(openIdConnectScheme)
+               .Configure<IConfiguration, IHttpClientFactory, IHostEnvironment>((options, configuration, httpClientFactory, hostEnvironment) =>
+               {
+                   options.Backchannel = httpClientFactory.CreateClient(KeycloakBackchannel);
+                   options.Authority = GetAuthorityUri(serviceName, realm);
+                   options.SignInScheme = cookieScheme;
+                   options.SignOutScheme = openIdConnectScheme;
```

Comment:
```suggestion
```

There's a reason the `OpenIdConnectHandler` [defaults to the `SignOutScheme` matching the `SignInScheme`](https://github.com/dotnet/aspnetcore/blob/28481ab0d6a31883a6c058d045ca8f72591a7eca/src/Security/Authentication/OpenIdConnect/src/OpenIdConnectPostConfigureOptions.cs#L42). It's used when the OIDC provider is the one [triggering the signout of our application](https://github.com/dotnet/aspnetcore/blob/28481ab0d6a31883a6c058d045ca8f72591a7eca/src/Security/Authentication/OpenIdConnect/src/OpenIdConnectHandler.cs#L199-L200), so presumably the user has already been signed out by the provider. We just want to delete the auth cookie.

Reply:
Done!